python/tests/test_collections.py

```diff
@@ -0,0 +1,312 @@
+from unittest.mock import Mock
+
+import pytest
+from numerous import collection
+from numerous._client import Client
+from numerous.collection.numerous_document import NumerousDocument
+from numerous.generated.graphql.client import Client as GQLClient
+from numerous.generated.graphql.collection_create import CollectionCreate
+from numerous.generated.graphql.collection_document import CollectionDocument
+from numerous.generated.graphql.collection_document_delete import (
+    CollectionDocumentDelete,
+)
+from numerous.generated.graphql.collection_document_set import CollectionDocumentSet
+from numerous.generated.graphql.collection_document_tag_add import (
+    CollectionDocumentTagAdd,
+)
+from numerous.generated.graphql.collection_document_tag_delete import (
+    CollectionDocumentTagDelete,
+)
+
+
+ORGANIZATION_ID = "test_org"
+COLLECTION_NAME = "test_collection"
+NESTED_COLLECTION_ID = "nested_test_collection"
+COLLECTION_REFERENCE_KEY = "test_key"
+COLLECTION_REFERENCE_ID = "test_id"
+NESTED_COLLECTION_REFERENCE_KEY = "nested_test_key"
+NESTED_COLLECTION_REFERENCE_ID = "nested_test_id"
+COLECTION_DOCUMNET_KEY = "test_document"
+
+
+def _collection_create_collection_reference(key: str, ref_id: str) -> CollectionCreate:
+    return CollectionCreate.model_validate(
+        {"collectionCreate": {"typename__": "Collection", "key": key, "id": ref_id}}
+    )
+
+
+def _collection_document_set_reference(key: str) -> CollectionDocumentSet:
+    return CollectionDocumentSet.model_validate(
+        {
+            "collectionDocumentSet": {
+                "__typename": "CollectionDocument",
+                "id": "2b72c972-d7c3-4864-91ce-6f65a9346202",
+                "key": key,
+                "data": "eyJ0ZXN0IjogInRlc3QifQ==",
+                "tags": [],
+            }
+        }
+    )
```

Comment:
In relation to another comment I made, where I ask about the test data being a bit hard to understand because it is not clear what it is (since it is not colocated with the test, and because it is base64-encoded), we could do something like the following.

It would make it possible to assert something like

```python
doc = collection.document("document-key")
assert doc.get() == DOCUMENT_DATA
```

```suggestion
DOCUMENT_DATA = {"test": "test"}
DOCUMENT_DATA_BASE64 = b64encode.encode(json.dumps(DOCUMENT_DATA))

def _collection_document_set_reference(key: str) -> CollectionDocumentSet:
    return CollectionDocumentSet.model_validate(
        {
            "collectionDocumentSet": {
                "__typename": "CollectionDocument",
                "id": "2b72c972-d7c3-4864-91ce-6f65a9346202",
                "key": key,
                "data": DOCUMENT_DATA_BASE64,
                "tags": [],
            }
        }
    )
```